cf/data/data.py

```diff
@@ -3831,304 +3831,70 @@ def _binary_operation(self, other, method):
-        # then result will get filled with data and returned. If we're
-        # an augmented arithmetic assignment then we'll update self
-        # with this new metadata.
-        # ------------------------------------------------------------
+        # Parse units
+        data0, other, new_Units = data0._combined_units(other, method, True)
 
-        result = data0.copy()
-        result._shape = new_shape
-        result._ndim = new_ndim
-        result._size = new_size
-        result._axes = new_axes
+        # Cast as dask arrays
+        dx0 = data0._get_dask()
+        dx1 = other._get_dask()
 
-        # ------------------------------------------------------------
-        # Set the data-type of the result
-        # ------------------------------------------------------------
-        if method_type in ("_eq", "_ne", "_lt", "_le", "_gt", "_ge"):
-            new_dtype = np.dtype(bool)
+        # Set if applicable the tolerance levels for the result
+        if method in ("__eq__", "__ne__"):  # what about l/g-t-e (x4)?
             rtol = self._rtol
             atol = self._atol
-        else:
-            if "true" in method:
-                new_dtype = np.dtype(float)
-            elif not inplace:
-                new_dtype = np.result_type(data0.dtype, other.dtype)
-            else:
-                new_dtype = data0.dtype
-        # --- End: if
 
         # ------------------------------------------------------------
-        # Set flags to control whether or not the data of result and
-        # self should be kept in memory
+        # Perform the binary operation with data0 (self) and data1 (other)
         # ------------------------------------------------------------
-        config = data0.partition_configuration(readonly=not inplace)
-
-        original_numpy_seterr = np.seterr(**_seterr)
-
-        # Think about dtype, here.
-
-        for partition_r, partition_s in zip(
-            result.partitions.matrix.flat, data0.partitions.matrix.flat
-        ):
-
-            partition_s.open(config)
-
-            indices = partition_s.indices
-
-            array0 = partition_s.array
-
-            if broadcasting:
-                indices = tuple(
-                    [
-                        (index if not broadcast_index else broadcast_index)
-                        for index, broadcast_index in zip(
-                            indices[align_offset:], broadcast_indices
-                        )
-                    ]
-                )
-                indices = (Ellipsis,) + indices
-
-            array1 = other[indices].array
-
-            # UNRESOLVED ISSUE: array1 could be much larger than the
-            # chunk size.
-
-            if not inplace:
-                partition = partition_r
-                partition.update_inplace_from(partition_s)
+        try:
+            if method == "__eq__":
+                result = _numpy_isclose(dx0, dx1, rtol=rtol, atol=atol)
```

Comment:
I'm think that for v4.0.0 we should think about not subverting `==` and `!=` with tolerance, and use our `isclose` in other parts of the code instead, in the interests of consistency with numpy, etc.

This would obvioulsy be an API change, so is not for consideration in this PR!

Reply:
As discussed and agreed just now in person, we will indeed reserve (not) equals operations for strict equality checking, and use `isclose` for operations within tolerances (and our `isequals` will use `isclose` type comparisons).

I'll shortly put up a new issue to list some API changes we are planning, that we can add to as we decide on them, and I'll include this as one on the initial listing. Thanks!